src/test/java/teammates/test/cases/BaseTestCaseWithBackDoorApiAccess.java

```diff
@@ -10,26 +10,50 @@
 import teammates.common.datatransfer.attributes.FeedbackSessionAttributes;
 import teammates.common.datatransfer.attributes.InstructorAttributes;
 import teammates.common.datatransfer.attributes.StudentAttributes;
+import teammates.common.util.Const;
+import teammates.common.util.ThreadHelper;
 import teammates.test.driver.BackDoor;
 
 /**
  * Base class for all test cases which are allowed to access the Datastore via {@link BackDoor}.
  */
 public abstract class BaseTestCaseWithBackDoorApiAccess extends BaseTestCaseWithDatastoreAccess {
 
+    private static final int BACKDOOR_GET_RETRY_COUNT = 100;
```

Comment:
I can get up to 20 retries while testing on my staging server haha. I think the live server would suffer even more.